mdsobjects/cpp/mdstreeobjects.cpp

```diff
@@ -1661,7 +1681,360 @@ StringArray *TreeNodeArray::getUsage()
 
 
 ////////////////End TreeNodeArray methods/////////////////////
+////////////////TreeNodeThinClient methods ///////////////////
+
+static const char *convertNciItm(int nciItm)
+{
+    switch(nciItm) {
+      case NciTIME_INSERTED: 		return "\'TIME_INSERTED\'";
+      case NciOWNER_ID: 		return "\'OWNER_ID\'"; 
+      case NciCLASS:			return "\'CLASS\'";
+      case NciDTYPE:			return "\'DTYPE\'";
+      case NciLENGTH:			return "\'LENGTH\'";
+      case NciSTATUS:			return "\'STATUS\'";
+      case NciCONGLOMERATE_ELT:		return "\'CONGLOMERATE_ELT\'";
+      case NciGET_FLAGS:		return "\'GET_FLAGS\'";
+      case NciNODE_NAME:		return "\'NODE_NAME\'";
+      case NciPATH:			return "\'PATH\'";
+      case NciDEPTH:			return "\'DEPTH\'";
+      case NciPARENT:			return "\'PARENT\'";
+      case NciBROTHER:			return "\'BROTHER\'";
+      case NciMEMBER:			return "\'MEMBER\'";
+      case NciCHILD:			return "\'CHILD\'";
+      case NciPARENT_RELATIONSHIP:	return "\'PARENT_RELATIONSHIP\'";
+      case NciK_IS_CHILD:		return "\'IS_CHILD\'";
+      case NciK_IS_MEMBER:		return "\'IS_MEMBER\'";
+      case NciCONGLOMERATE_NIDS:	return "\'CONGLOMERATE_NIDS\'";
+      case NciORIGINAL_PART_NAME:	return "\'ORIGINAL_PART_NAME\'";
+      case NciNUMBER_OF_MEMBERS:	return "\'NUMBER_OF_MEMBERS\'";
+      case NciNUMBER_OF_CHILDREN :	return "\'NUMBER_OF_CHILDREN\'";
+      case NciMEMBER_NIDS:		return "\'MEMBER_NIDS\'";
+      case NciCHILDREN_NIDS:		return "\'CHILDREN_NIDS\'";
+      case NciFULLPATH:			return "\'FULLPATH\'";
+      case NciMINPATH:			return "\'MINPATH\'";
+      case NciUSAGE:			return "\'USAGE\'";
+      default:				std::cout << "Unecpected NCI item in TreeNodeThinClient" << std::endl; return "\'\'";
```

Comment:
typo: unexpected. Also, should the library be doing std::cout, or throwing an exception?